python/ray/tune/trial_runner.py

```diff
@@ -1127,7 +1129,8 @@ def _process_trial_save(self, trial):
                     trial=trial,
                     checkpoint=trial.saving_to)
                 trial.on_checkpoint(trial.saving_to)
-                self.trial_executor.try_checkpoint_metadata(trial)
+                if trial.checkpoint.storage != Checkpoint.MEMORY:
+                    self.trial_executor.mark_trial_to_checkpoint(trial)
```

Comment:
why is this functionality not part of runner?

Reply:
I agree, that should be moved into the runner eventually